ocaml/testsuite/tests/typing-local/local.ml

```diff
@@ -777,6 +777,45 @@ Line 2, characters 20-45:
 Error: This function is local-returning, but was expected otherwise
 |}]
 
+(*
+ * Modification of parameter modes in argument position
+ *)
+
+let use_local (local_ f : _ -> _ -> _) x y =
+  f x y
+let use_global (f : _ -> _ -> _) x y = f x y
+
+let foo x y = x +. y
+let bar (local_ x) (local_ y) = let _ = x +. y in ()
+
+let result = use_local foo 1. 2.
+[%%expect{|
+val use_local : local_ ('a -> 'b -> 'c) -> 'a -> 'b -> 'c = <fun>
```

Comment:
The high-level point of this PR is that `use_local` will be able to accept both `foo` and `bar`. This tests that this is the case. `use_global` is there to check that it doesn't work in the case that would be unsafe -- where it might leak the partially applied version.